internal/clients/elasticsearch/index.go

```diff
@@ -544,3 +544,158 @@ func DeleteIngestPipeline(ctx context.Context, apiClient *clients.ApiClient, nam
 	}
 	return diags
 }
+
+func PutTransform(ctx context.Context, apiClient *clients.ApiClient, transform *models.Transform, params *models.PutTransformParams) diag.Diagnostics {
+	fmt.Println("entering PutTransform for", transform.Name)
+	var diags diag.Diagnostics
+	transformBytes, err := json.Marshal(transform)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	fmt.Printf("%s\n", transformBytes)
+
+	esClient, err := apiClient.GetESClient()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	putOptions := []func(*esapi.TransformPutTransformRequest){
+		esClient.TransformPutTransform.WithContext(ctx),
+		esClient.TransformPutTransform.WithDeferValidation(params.DeferValidation),
+		esClient.TransformPutTransform.WithTimeout(params.Timeout),
+	}
+
+	res, err := esClient.TransformPutTransform(bytes.NewReader(transformBytes), transform.Name, putOptions...)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	defer res.Body.Close()
+	if diags := utils.CheckError(res, fmt.Sprintf("Unable to create transform: %s", transform.Name)); diags.HasError() {
+		return diags
+	}
+
+	if params.Enabled {
+		startOptions := []func(*esapi.TransformStartTransformRequest){
+			esClient.TransformStartTransform.WithContext(ctx),
+			esClient.TransformStartTransform.WithTimeout(params.Timeout),
+		}
+		_, err := esClient.TransformStartTransform(transform.Name, startOptions...)
```

Comment:
We should be checking for an error response here as well. Similar to the `utils.CheckError` call on ln 575.